examples/nodejs/rate-limiter/README.md

```diff
@@ -0,0 +1,109 @@
+[![project stability](https://momentohq.github.io/standards-and-practices/badges/project-stability-stable.svg)](https://github.com/momentohq/standards-and-practices/blob/main/docs/momento-on-github.md)
+
+## Why rate-limiters?
+
+The need for rate-limiting stems from the fundamental requirement to maintain the health and quality of any service. Without it, resources could easily become overwhelmed, leading to service degradation or outright failure. This is particularly important in distributed systems and web services where the client requests can vary dramatically in volume and frequency. Rate-limiting ensures a fair distribution of resources, prevents abuse, and can even be a crucial component in defending against certain types of cyber-attacks, such as Distributed Denial of Service (DDoS) attacks.
+
+Some common use-cases of rate-limiting includes:
+
+- API Management: In a platform offering various APIs, rate-limiting is crucial to prevent a single user or service from monopolizing the bandwidth, ensuring that all users have equitable access to the resources.
+
+- E-commerce Websites: During high-traffic events like Black Friday sales, rate-limiting can prevent the website from crashing by controlling the influx of user requests, thus providing a stable and fair shopping experience to all customers.
+
+- Online Gaming Servers: Rate-limiting can help in mitigating cheating by throttling the number of actions a player can perform in a given time, ensuring a level playing field and maintaining the game's integrity.
+
+## Getting started
+
+We provide a `MomentoRateLimiter` class that uses Momento's `increment` and `updateTTL` [APIs](https://docs.momentohq.com/cache/develop/api-reference) to achieve rate-limiting.
+Incorporating the `MomentoRateLimiter` class into your application is a straightforward process. Begin by setting up a Momento cache client, then link it to the rate-limiter, specifying the user-specific limits. This rate-limiter operates with minute-level precision, meaning that the defined limits will be enforced on a per-minute basis for each user or entity.
+
+To get started with the rate-limiter:
+- You will need a Momento API key. You can obtain one from the [Momento Console](https://console.gomomento.com).
+- You will need to create a cache called `rate-limiter` from the console as well!
+
+Once you have the key and the cache created, you can begin integration! Remember to store your API key in an environment variable named `MOMENTO_API_KEY`.
+
+```typescript
+const momento = await CacheClient.create({
+  configuration: Configurations.Laptop.v1(),
+  credentialProvider: CredentialProvider.fromEnvironmentVariable({
+    environmentVariableName: "MOMENTO_API_KEY",
+  }),
+  defaultTtlSeconds: 6000,
+});
+
+const tpmLimit = 10;
+const rateLimiter = new MomentoRateLimiter(momento, tpmLimit);
```

Comment:
would be nice to accept cache name as an argument here rather than hard-coding in the lib, but not a huge deal.

Reply:
When we spoke to Jah I got the feeling that most customers like to decouple control and data plane so left it as is

Reply:
that's different from being locked in to a hard coded cache name though :)

this is okay for example code where we expect users to copy/paste/modify. this would not be okay in a library though :)

Reply:
ah yeah gotcha now!